### PR TITLE
Collections cleanup

### PR DESCRIPTION
**Pull request by frbncis**
I went through System.Collections.* and removed dead code as per #17905 

* [System.Collections.Concurrent](http://tempcoverage.blob.core.windows.net/report4/System.Collections.Concurrent.diff.html) ([csv](http://tempcoverage.blob.core.windows.net/report4/System.Collections.Concurrent.diff.csv)) - nothing to clean up
* [System.Collections](http://tempcoverage.blob.core.windows.net/report4/System.Collections.diff.html) ([csv](http://tempcoverage.blob.core.windows.net/report4/System.Collections.diff.csv)) - Cleaned up, as per #32654, #32676
* [System.Collections.Immutable](http://tempcoverage.blob.core.windows.net/report4/System.Collections.Immutable.diff.html) ([csv](http://tempcoverage.blob.core.windows.net/report4/System.Collections.Immutable.diff.csv)) - Nothing to clean up
* [System.Collections.NonGeneric](http://tempcoverage.blob.core.windows.net/report4/System.Collections.NonGeneric.diff.html) ([csv](http://tempcoverage.blob.core.windows.net/report4/System.Collections.NonGeneric.diff.csv)) - unused strings
* [System.Collections.Specialized](http://tempcoverage.blob.core.windows.net/report4/System.Collections.Specialized.diff.html) ([csv](http://tempcoverage.blob.core.windows.net/report4/System.Collections.Specialized.diff.csv)) - ~~cleaned up~~ nothing to remove

### CODE REVIEW

File: src/Common/src/CoreLib/System/Collections/CompatibleComparer.cs [hunk 1]
```diff
@@ -17,10 +17,6 @@ internal CompatibleComparer(IHashCodeProvider hashCodeProvider, IComparer compar
             _comparer = comparer;
         }
 
-        internal IHashCodeProvider HashCodeProvider => _hcp;
-
-        internal IComparer Comparer => _comparer;
-
```
Comment (stephentoub) [inline src/Common/src/CoreLib/System/Collections/CompatibleComparer.cs]: These need to be put back. This code is shared with CoreLib in the coreclr repo, and these properties are used there.

File: src/System.Collections/src/System/Collections/Generic/LinkedList.cs [hunk 2]
```diff
@@ -549,11 +549,6 @@ internal Enumerator(LinkedList<T> list)
                 _index = 0;
             }
 
-            private Enumerator(SerializationInfo info, StreamingContext context)
-            {
-                throw new PlatformNotSupportedException();
-            }
-
```
Comment (stephentoub) [inline src/System.Collections/src/System/Collections/Generic/LinkedList.cs]: @ViktorHofer, this type isn't marked as [Serializable]. I assume the interfaces remained for public API compatibility, but there's no issue removing this private ctor, right?

**Comment by stephentoub** [comment]
Thanks for working on this.

File: src/System.Collections/src/System/Collections/Generic/LinkedList.cs (hunk 2 above)
Comment (ViktorHofer) [reply src/System.Collections/src/System/Collections/Generic/LinkedList.cs]: Yeah that can be safely removed :)

File: src/Common/src/CoreLib/System/Collections/CompatibleComparer.cs (hunk 1 above)
Comment (frbncis) [reply src/Common/src/CoreLib/System/Collections/CompatibleComparer.cs]: My mistake, should be back now.

**Pull request merged**